util/multirun.sh

```diff
@@ -35,27 +67,78 @@ fi
 
 cip="$1"
 shift
+args=("$@")
 
-for opts in "$@"; do
-    manifest=$(echo "$opts" | cut -d, -f1)
+for arg in "${args[@]}"; do
+    if ! [[ "$arg" =~ .*,.* ]]; then
+        echo >&2 "invalid argument: $arg"
+        usage
+        exit 1
+    fi
+done
+
+# If we only want to run against changed manifest files, then filter out those
+# files that were not changed since the last commit.
+if [[ "${CIP_GIT_REV:-}" ]]; then
+    if [[ ! -d "${CIP_GIT_DIR:-}" ]]; then
+        echo >&2 "CIP_GIT_DIR not set (must be set if CIP_GIT_REV is set)"
+        usage
+        exit 1
+    fi
+    pushd "${CIP_GIT_DIR}"
+    changed_files=()
+    while IFS= read -r f; do
+        changed_files+=( "$f" )
+    done < <( git diff-tree --no-commit-id --name-only -r "${CIP_GIT_REV}" )
```

Comment:
I don't know how this is managing the git checkout so I can't say it's right or not - comments might help the casual reader understand

Reply:
Will add some comments here.